### PR TITLE
Dim toolbar icon on unstyled tabs

### LINKED ISSUE #100: Make icon act like Firefox
White icon when there's no styles.

https://developer.chrome.com/extensions/browserAction#method-setIcon

### PR DESCRIPTION
**Pull request by tophf**
Closes #100

* **Unstyled tab icon** is dimmed: <br>`S` remains the same, bg = 20%, outer border = 50% opacity. Transparent background should look good on non-bright themes, supposedly, unlike 100% white bg that would have been brighter than the default bg.<br>

* **in *disableAll* mode** the icon is also dimmed, style count badge is displayed on a lightgray background #aaa - should we have an option to customize this color to make it look good with dark themes?

* The default 19x19 icon was recreated using SVG vector source: <br>`S` and its outline were straightened a bit (check it out using GitHub's "Swipe" preview mode in diff)
* 38x38 retina-aware icons were added. Not tested.
* All icons were compressed with `pngquant --quality 50-80`

* Default icon blinks for an instant here (usually less than 0.1s) when I navigate a styled site, this is what happens:
  1. onBeforeNavigate: `updateIcon`: OK
  2. Chrome resets the icon on a whim<br>(particularly wacky when you navigate from an unstyled page to a styled one)
  3. onCommitted: `updateIcon`: OK

  The whim part [apparently](https://developer.chrome.com/extensions/webNavigation#event_order) is incurable as there's nothing between onBeforeNavigate & onCommitted:

    >events are fired in the following order:
    >`onBeforeNavigate -> onCommitted -> onDOMContentLoaded -> onCompleted`

  Although we can leave the colored icon as default in manifest.json and not care about it being incorrect on about:blank and other chrome:// pages except for Edit and Manage.

  Other extensions are updating their icons after ~1 second or so, which makes them look not too smart in comparison.

### CODE REVIEW

**Comment by tophf** [comment]
Oops, there is a problem: the default icon is showing briefly when you navigate a styled site, looks fidgety and messy.

**Comment by tophf** [comment]
Kind of fixed, see the notes above.

File: background.js [hunk 1]
```diff
@@ -11,19 +12,21 @@ function webNavigationListener(method, data) {
 		return;
 	}
 	getStyles({matchUrl: data.url, enabled: true, asHash: true}, function(styleHash) {
-		chrome.tabs.sendMessage(data.tabId, {method: method, styles: styleHash});
-		// Don't show the badge for frames
-		if (data.frameId == 0 && prefs.getPref("show-badge")) {
```
Comment (tophf) [inline background.js]: data.frameId is 0 here because of line 12.

**Comment by JasonBarnabe** [comment]
I've put the logo source at https://userstyles.org/newlogo.xcf . Not sure if that's helpful at this point.

**Comment by tophf** [comment]
To create the vector I've previously used a slightly distorted 128px icon as a source, so now I've fixed a few discrepancies from the real source and realigned all the icons on pixel grid.

**Comment by JasonBarnabe** [comment]
It works inconsistently when enabling a style on the new tab page.

**Comment by tophf** [comment]
* Fixed.
* replaced that pesky `try{}` in `background.js::chrome.contextMenus.create` with `chrome.runtime.lastError`.

**Comment by tophf** [comment]
Rebased on master.

**Comment by JasonBarnabe** [comment]
With two tabs and one global style:

- Enable the style - icon is colourful, shows 1 in red box
- Turn all styles off - icon is grey, shows 1 in grey box
- Switch tabs - icon is grey, shows 1 in grey box
- Turn all styles on - icon is colourful, shows 1 in **grey** box

**Comment by tophf** [comment]
Fixed, hopefully.
However out of about 50 repetitions I was able to reproduce this only once in the very beginning.

**Pull request merged**